core/src/main/java/org/infinispan/marshall/protostream/impl/SerializationContextRegistry.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.marshall.protostream.impl;
+
+import org.infinispan.protostream.BaseMarshaller;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.protostream.config.Configuration;
+
+/**
+ * Manages {@link SerializationContext} across modules for use by various components.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+public interface SerializationContextRegistry {
+
+   void addProtoFile(MarshallerType type, FileDescriptorSource fileDescriptorSource);
```

Comment:
This method is never used, do we want to keep it?

Reply:
I think we'll need this when converting the GlobalMarshaller, so I added it for completeness. I figured as it's an internal interface, adding/removing methods would be ok as required.

Reply:
It's actually required in my latest commit, but I have removed the `addConfiguration` method.